### PR TITLE
feat(services): created a videoplayer service to load script and embed

### PR DESCRIPTION
**Pull request by mkothur**
### Related Ticket(s)

Services - #1065  
### Description

Create a service to add a script tag dynamically to load the Kaltura video player metadata and embed the video for a placeholder by giving the target ID.
NOTE: for testing, purpose invoked the service embed method in the LightboxMediaViewer component.

### Changelog

*
**Changed**

Created a service in `packages/services`
Calling the async method in `packages/react`
image

### CODE REVIEW

File: packages/services/.env.example [hunk 1]
```diff
@@ -15,3 +15,7 @@ CORS_PROXY=<CORS proxy server for fetching from www.ibm.com, e.g. https://yourpr
 
 # Analytics - scroll
 SCROLL_TRACKING=<boolean to enable scroll tracking, e.g. false>
+
+# Kaltura Video Player
+KALTURA_PARTNER_ID=243342
+KALTURA_UICONF_ID=12905712
```
Comment (jeffchew) [inline packages/services/.env.example]: @mkothur shouldn't include the actual numbers in the `.env.example`, should be more of a description like above. We probably should only share these numbers internally and not in the source.

File: packages/react/src/components/LightboxMediaViewer/LightboxMediaViewer.js [hunk 2]
```diff
@@ -31,6 +32,15 @@ const { prefix } = settings;
  * @returns {*} JSX Object
  */
 const LightboxMediaViewer = ({ title, image, copy, ...modalProps }) => {
+  useEffect(() => {
+    (async () => {
+      const embedData = await VideoPlayerAPI.embedVideo(
+        '0_uka1msg4',
+        document.getElementById('kaltura_player')
+      );
+      console.log(embedData);
```
Comment (jeffchew) [inline packages/react/src/components/LightboxMediaViewer/LightboxMediaViewer.js]: Shouldn't be hard coding this in the component. This story should just be capturing the creation of the service, so we shouldn't include any changes to this component here.

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
+const _partnerId = process.env.KALTURA_PARTNER_ID || 243342;
+const _uiConfId = process.env.KALTURA_UICONF_ID || 12905712;
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: Add JSDocs here. Should make some mention that these values are for production for IBM.com.

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+let _attempt = 0;
+
+/**
+ *
+ * Tracks the script either in loaded state or loading state
+ * @type {boolean}
+ * @private
+ */
+let _scriptLoaded = false;
+let _scriptLoading = false;
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: Add JSDocs.

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
+const _partnerId = process.env.KALTURA_PARTNER_ID || 243342;
+const _uiConfId = process.env.KALTURA_UICONF_ID || 12905712;
+
+const _embedUrl = `https://cdnapisec.kaltura.com/p/${_partnerId}/sp/${_partnerId}00/embedIframeJs/uiconf_id/${_uiConfId}/partner_id/${_partnerId}`;
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: Add JSDocs.

File: packages/services/.env.example (hunk 1 above)
Comment (mkothur) [reply packages/services/.env.example]: Okay @jeffchew

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+   * @param {string} videoId  The videoId we're embedding the placeholder for.
+   * @param {string} targetId The targetId the ID where we're putting the placeholder.
+   * @returns {object}  object
+   */
+  static async embedVideo(videoId, targetId) {
+    let kWidget;
+    return await this.checkScript().then(() => {
+      if (typeof kWidget == 'undefined') {
+        return;
+      }
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: the `embedVideo` logic shouldn't be capturing this. This should theoretically already be handled by the `checkScript`.

File: packages/react/src/components/LightboxMediaViewer/LightboxMediaViewer.js (hunk 2 above)
Comment (mkothur) [reply packages/react/src/components/LightboxMediaViewer/LightboxMediaViewer.js]: yeah, correct for testing purpose I have added @jeffchew

File: packages/services/src/services/VideoPlayer/VideoPlayer.js (hunk 3 above)
Comment (mkothur) [reply packages/services/src/services/VideoPlayer/VideoPlayer.js]: Sure

File: packages/services/src/services/VideoPlayer/VideoPlayer.js (hunk 4 above)
Comment (mkothur) [reply packages/services/src/services/VideoPlayer/VideoPlayer.js]: Okay

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        entry_id: videoId,
+        flashvars: {
+          autoPlay: false,
+        },
+        // Ready callback is issued for this player:
+        readyCallback: function(playerId) {
+          console.log('kWidget player ready: ' + playerId, targetId);
+          var kdp = document.getElementById(playerId);
+          kdp.kBind('Play', function() {
+            console.log('do Play called on  ' + playerId);
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: I know this is still in draft, but should be mindful on what console logs we still have here...

File: packages/services/src/services/VideoPlayer/VideoPlayer.js (hunk 6 above)
Comment (mkothur) [reply packages/services/src/services/VideoPlayer/VideoPlayer.js]: Yeah I have updated in `checkScript`

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  /**
+   * Gets the embed meta data
+   *
+   * @param {string} videoId  The videoId we're embedding the placeholder for.
+   * @param {string} targetId The targetId the ID where we're putting the placeholder.
+   * @returns {object}  object
+   */
+  static async embedVideo(videoId, targetId) {
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: I see the method for `embedVideo`, we also need a method for `api` to make the `kWidget.api` call.

File: packages/styles/scss/components/lightbox-media-viewer/_lightbox-media-viewer.scss [hunk 9]
```diff
@@ -34,6 +40,10 @@
         color: $text-02;
       }
     }
+    &__videoPlayer {
+      width: '400px';
+      height: '400px';
+    }
```
Comment (jeffchew) [inline packages/styles/scss/components/lightbox-media-viewer/_lightbox-media-viewer.scss]: Let's not include any coding for the embed for this story.

File: packages/services/src/services/VideoPlayer/VideoPlayer.js (hunk 8 above)
Comment (mkothur) [reply packages/services/src/services/VideoPlayer/VideoPlayer.js]: yeah Jeff, will add `api` method too.

File: packages/styles/scss/components/lightbox-media-viewer/_lightbox-media-viewer.scss (hunk 9 above)
Comment (mkothur) [reply packages/styles/scss/components/lightbox-media-viewer/_lightbox-media-viewer.scss]: sorry for testing purpose I have added these, will remove them in a final PR.

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+/**
+ *
+ * Tracks the script either in loaded state or loading state
+ * @type {boolean} _scriptLoaded to track the script loaded or not
+ * * @type {boolean} _scriptLoading to track the script loading or not
+ * @private
+ */
+let _scriptLoaded = false;
+let _scriptLoading = false;
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: @mkothur can you add two separate JSDoc comments for `_scriptLoaded` and `_scriptLoading`?

File: packages/services/src/services/VideoPlayer/VideoPlayer.js [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+  }
+
+  /**
+   * Gets the api data
+   *
+   * @param {string} videoId  The videoId we're embedding the placeholder for.
+   * @returns {object}  object
+   */
+  static async api(videoId) {
+    return await new kWidget.api({ wid: _partnerId }).doRequest(
```
Comment (jeffchew) [inline packages/services/src/services/VideoPlayer/VideoPlayer.js]: @mkothur should wrap this with the `checkScript` as well:
```
return await this.checkScript().then(() => {
   ... call kWidget.api and return the response...
}
```

Comment (mkothur) [reply packages/services/src/services/VideoPlayer/VideoPlayer.js]: updated @jeffchew

File: packages/services/src/services/VideoPlayer/VideoPlayer.js (hunk 10 above)
Comment (mkothur) [reply packages/services/src/services/VideoPlayer/VideoPlayer.js]: done!

**Comment by jeffchew** [comment]
@mkothur fyi one other thing we're missing here are unit tests.

**Comment by mkothur** [comment]
@jeffchew will add test suits.

**Comment by jeffchew** [comment]
> @jeffchew will add test suits.

@mkothur let's actually get the unit tests in a separate PR, this PR is getting a bit stale.

**Comment by jeffchew** [comment]
Issue created to add unit tests: https://github.com/carbon-design-system/ibm-dotcom-library/issues/1323

**Comment by mkothur** [comment]
Sure @jeffchew will add. test cases.

**Pull request merged**